src/components/List.jsx

```diff
@@ -1,23 +1,57 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import db from '../lib/firebase';
-import { collection } from 'firebase/firestore';
+import { collection, doc, updateDoc } from 'firebase/firestore';
 import { useCollection } from 'react-firebase-hooks/firestore';
+import moment from 'moment';
 
 export const List = ({ token }) => {
   const [value, loading, error] = useCollection(collection(db, token), {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
 
+  const [itemId, setItemId] = useState(' ');
+
+  const [checkedItems, setCheckedItems] = useState({});
+
+  const handleChange = (data, e) => {
+    setCheckedItems({ ...checkedItems, [e.target.name]: e.target.value });
+
+    setItemId(data.id);
+  };
+
+  const docRef = doc(db, token, itemId);
+  console.log(docRef);
+
+  useEffect(() => {
+    console.log('Hello I am here, and I am listening to youuuuu');
+    const updateDocument = async () => {
+      await updateDoc(docRef, {
+        purchased_date: moment().format(),
+      });
+    };
+    return () => updateDocument();
+  }, []);
+
   return (
     <div>
       <h4> Shareable List Token : {token} </h4>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
       Collection
       {value && (
-        <ul>
+        <ul className="collection-list">
           {value.docs.map((doc) => (
-            <li key={doc.id}>{doc.data().item}</li>
+            <li key={doc.id}>
+              <input
+                type="checkbox"
+                // checked={doc.data().purchased_date ? false : true}
+                name={doc.id}
+                id={doc.id}
+                value={doc.data().item}
+                onChange={(e) => handleChange(doc, e)}
```

Comment:
It may make more sense to use `onClick` event handler instead of `onChange`. In this case, you are waiting for the user to click on it to trigger saving to the database. An `onChange` and the `useEffect` above will run too many times.

Reply:
Thanks Yenly!